### PR TITLE
Added more context to invalid text string errors

### PR DESCRIPTION
**Pull request by EvanBacon**
## Summary

This PR is a companion to https://github.com/facebook/react-native/pull/32544 which was recommended here https://github.com/facebook/react-native/pull/32544#issuecomment-961837109. In React Native, the stack traces shown in LogBox often just point to fast refresh and don't provide much context into cases where a user adds a string outside of a Text element, this PR improves discoverability a little by adding the invalid text to the error message.

## How did you test this change?

Update the error message format to include the invalid text string.

### CODE REVIEW

**Comment by yungsters** [comment]
Thanks again for your contribution, @EvanBacon.

I'll copy over some of my feedback from https://github.com/facebook/react-native/pull/32544:

> tl;dr — I previously attempted a similar change, but was recommended to revert it because @acdlite and @sebmarkbage raised concerns with logging arbitrary strings which may be sensitive (e.g. passwords, banking information).

> @EvanBacon — What do you think about only appending the string if an opt-in flag is set (e.g. `global.debugInvalidTextStrings`), and then changing the error message to inform people to do that to opt into the extra error text? (And maybe when the opt-in flag is set, remind them to remove it when they're done debugging.)

It looks like the current status of this PR does not incorporate an opt-in flag as I previously suggested. Does this PR simply need to be updated, or do you have concerns about that approach?

**Comment by EvanBacon** [comment]
@gaearon perhaps the most minimal repro in native:

[blank.zip](https://github.com/facebook/react/files/7849744/blank.zip)

## Setup

- `yarn`
- `yarn start`
- Press `i` or `a` (opens in Expo Go)

### Code

`App.js`
```tsx
import { View } from "react-native";

export default function App() {
  return (
    <View
      style={{
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
      }}>
      Hey
    </View>
  );
}
```


## Error Output

```
Error: Text strings must be rendered within a <Text> component.

This error is located at:
    in RCTView (at View.js:34)
    in View (created by App)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:47)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in DevAppContainer (at AppContainer.js:122)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at renderApplication.js:40)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/ReactFiberErrorDialog.js:43:2 in showErrorDialog
at node_modules/react-native/Libraries/Core/setUpReactRefresh.js:43:6 in performReactRefresh
at node_modules/metro-runtime/src/polyfills/require.js:655:10 in <unknown>

Error: Text strings must be rendered within a <Text> component.

This error is located at:
    in RCTView (at View.js:34)
    in View (created by App)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:47)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:107)
    in DevAppContainer (at AppContainer.js:122)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:134)
    in AppContainer (at renderApplication.js:40)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:172:19 in handleException
at node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at node_modules/expo-error-recovery/build/ErrorRecovery.fx.js:12:21 in <unknown>
```

### Initial Load

Simulator Screen Shot - iPhone 13 Pro - 2022-01-11 at 13 41 36

### Fast Refresh

Simulator Screen Shot - iPhone 13 Pro - 2022-01-11 at 13 37 19
